server/repository/src/db_diesel/number_row.rs

```diff
@@ -9,26 +10,25 @@ use diesel::result::Error::NotFound;
 use diesel::{prelude::*, sql_query, sql_types::Text};
 
 use diesel::sql_types::BigInt;
-use diesel_derive_enum::DbEnum;
 
 table! {
     number (id) {
         id -> Text,
         value -> BigInt,
         store_id -> Text,
-        #[sql_name = "type"] type_ -> crate::db_diesel::number_row::NumberRowTypeMapping,
+        #[sql_name = "type"] type_ -> Text,
     }
 }
 
-#[derive(DbEnum, Debug, Clone, PartialEq, Eq)]
-#[DbValueStyle = "SCREAMING_SNAKE_CASE"]
+#[derive(AsExpression, Debug, Clone, PartialEq, Eq)]
 pub enum NumberRowType {
     InboundShipment,
     OutboundShipment,
     InventoryAdjustment,
     RequestRequisition,
     ResponseRequisition,
     Stocktake,
+    Program(String),
 }
```

Comment:
The way it is implemented, this enum should go into the service layer since the repo only works with a string. The "from_string" and "to_string" methods should ideally be next to each other.

However, I think it would also be fine to have the enum in the repository layer and the repository layer decides how to store it. For example, an alternative NumberRowRepository implementation could decide to use the initial approach to have a custom number row without affecting the service layer.

Think this is similar to what @andreievg meant in his comment?

Reply:
Yes, I think that is what he meant, started working on this idea.
Part of the challenge is most of the uses for numbers interact directly on the repository layer not the service. Another thought I had just now is to move more of the usages to the service layer. Like you say if they were using the service layer, the underlying repository shouldn't matter.

Reply:
@clemens-msupply I've actually made a change more back the other way, taking `NumberRowType` for everything except for functions that need to use `NumberRow` directly. 
Are ok with the latest updates? 
I currently think it's probably the right balance with the time invested and a useful solution.